Dockerfile

```diff
@@ -0,0 +1,53 @@
+# Build compilation image
+FROM ruby:3.0.2-alpine3.13 as builder
+
+# The application runs from /app
+WORKDIR /app
+
+# build-base: compilation tools for bundle
+# git: used to pull gems from git
+# yarn: node package manager
+RUN apk add --update --no-cache build-base git postgresql-dev curl-dev shared-mime-info tzdata && \
+  cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
+  echo "Europe/London" > /etc/timezone
+
+# Install bundler to run bundle exec
+# This should be the same version as the Gemfile.lock
+RUN gem install bundler:2.2.18 --no-document
+RUN bundle config set without 'development test'
+
+# Install gems defined in Gemfile
+COPY .ruby-version Gemfile Gemfile.lock /app/
+RUN bundle install --jobs=4 --no-binstubs
+
+# Copy all files to /app (except what is defined in .dockerignore)
+COPY . /app/
+
+
+# Cleanup to save space in the production image
+RUN rm -rf node_modules log tmp && \
+      rm -rf /usr/local/bundle/cache && \
+      rm -rf .env && \
+      find /usr/local/bundle/gems -name "*.c" -delete && \
```

Comment:
Can you achieve this in one with a regex?

Reply:
Probably, but I think it clearer this way, as each line is targeting a specif "left over". 
Might not be worth the time getting it right, and there will be some risk.